components/push/android/src/main/java/mozilla/appservices/push/PushManager.kt

```diff
@@ -63,11 +62,12 @@ class PushManager(
         channelID: String,
         scope: String
     ): SubscriptionResponse {
-        val json = rustCallForString { error ->
+        val respBuffer = rustCallForBuffer { error ->
             LibPushFFI.INSTANCE.push_subscribe(
                 this.handle.get(), channelID, scope, error)
         }
-        return SubscriptionResponse.fromString(json)
+        val response = MsgTypes.SubscriptionResponse.parseFrom(respBuffer)
```

Comment:
@eoger Hi! While running unit tests on my dev machine, I get errors at this line parsing the returned buffer into a auto generated message object. The error message is: Protocol message contained an invalid tag (zero). I will keep working on this, and it would be great if you can point me to the mistakes I have made here (only if you have time)! Thanks!

Reply:
Hum I spent a half hour and couldn't find out what's wrong either. I wonder if the ByteArrays we get are correct on the Android side? Maybe the unit tests somehow override what's returned on the FFI border?

Reply:
Thanks for helping!! Good point, I haven't look into the test code very much yet. I will take a closer look after work today.

Reply:
I think the cause of the issue is that there is a missing encoding/decoding of the ByteArray returned by `rustCallForBuffer`. If I directly call `asCodedInputStream()` on the protocol buffer object, it works.

Reply:
Oh by bad I didn't realize you were using `getByteArray`, great find!

Reply:
Actually I realized it was un-used code sitting in master.